src/main/java/org/teragrep/JavaPcre.java

```diff
@@ -132,10 +166,45 @@ public void pcre2_versioncheck(){
 //        compile_options = new LibJavaPcre.OptionsStruct();
 //    }
 
-    public void pcre2_compile_java(String pat, int size){
+    public void pcre2_gcontext_create(){
+        gcontext = LibJavaPcre.INSTANCE.pcre2_gcontext_create();
+    }
+    public void pcre2_ccontext_create(){
+        if (gcontext == null){
+            LOGGER.error("General context is not initialized properly!");
```

Comment:
please consider using
```
throw new IllegalArgumentException("!enuff data")
```
to block usage when there are configuration mistakes

Reply:
Added exceptions to block usage with incorrect configurations.